### PR TITLE
[Updated] Feature/trade ship visit

### PR DESCRIPTION
**Pull request by sboigelot**
Allow to call trade ship from the landing pad (this is a temporary behavior to test the trade ship visit, a game event will be set later instead)

Added Trader Ship sprite (I'm not an artist, feel free to replace :D)
Added Trader Ship Controller allowing a trade ship to come from one end of the map, to the landing pad, trade with the player, then leave to the other end.

---The trading itself is not part of this PR.--- EDIT: it works in the PR now, except for the currency system, I will make another PR to link the currencies in the trade

ezgif com-optimize 1

### CODE REVIEW

**Comment by koosemose** [comment]
I absolutely love this. At some point in the future it may be useful to generalize this, so we can have ships come in for all sorts of purposes, then fire an action when they arrive at the selected destination (such as spawning new crewmembers, or spawning pirates for a raid... Could even potentially code a meteorite strike as a ship, just go to some point on the outer wall of the base and explode... though it'd probably need to go faster.

File: Assets/StreamingAssets/Data/Furniture.xml [hunk 1]
```diff
@@ -341,6 +340,7 @@
 
         <ContextMenuAction FunctionName="LandingPad_Test_ContextMenuAction" Text="Lua Test Deconstruct 1" RequiereCharacterSelected="false"/>
         <ContextMenuAction FunctionName="LandingPad_Test_ContextMenuAction" Text="Lua Test Deconstruct 2" RequiereCharacterSelected="true" />
```
Comment (vogonistic) [inline Assets/StreamingAssets/Data/Furniture.xml]: Now that we have proper context actions, you should remove the two test actions.

**Comment by vogonistic** [comment]
Did you draw the ship or do we need to add a license for it?

File: Assets/StreamingAssets/Data/Furniture.xml (hunk 1 above)
Comment (sboigelot) [inline Assets/StreamingAssets/Data/Furniture.xml]: I woudn't do that yet, it's still the only example of how to use it with character.

**Comment by sboigelot** [comment]
@vogonistic I draw it inspiring myself from google-image. But every pixel has been place by me, then I added some effect with Paint.net. No licence requiered.

**Comment by dustinwloring1988** [comment]
Nice cant wait to have this in the project.The possibilities are endless with this.

**Comment by Mizipzor** [comment]
This branch has conflicts that must be resolved, probably due to the huge StyleCop cleanup that just happened. Fix them and remove the `[Conflicts]` tag. :)

**Comment by Mizipzor** [comment]
Would it be possible to also do a stylecop pass on this? Seeing as we just completed the big cleanup. :)

**Comment by sboigelot** [comment]
Fixed merge conflics + style cop

**Comment by sboigelot** [comment]
I'll wait for saturday stylecop to end before fixing conflict again

**Comment by sboigelot** [comment]
Post saturday stylecop fix done

**Comment by dustinwloring1988** [comment]
Thanks cant wait

**Comment by sboigelot** [comment]
@dustinwloring1988 me too. I want to improve it more, but I'm afraid to put too much before the first merge.

File: Assets/StreamingAssets/Settings/Settings.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <worldWidth>101</worldWidth>
   <worldHeight>101</worldHeight>
   <localization>en_US</localization>
-  <DialogBoxSettings_langToggle>True</DialogBoxSettings_langToggle>
+  <DialogBoxSettings_langToggle>False</DialogBoxSettings_langToggle>
```
Comment (Grenkin1988) [inline Assets/StreamingAssets/Settings/Settings.xml]: Maybe this change should not be here

Comment (sboigelot) [inline Assets/StreamingAssets/Settings/Settings.xml]: @Grenkin1988 Reverted

**Comment by vogonistic** [comment]
This PR adds 30+ of StyleCop warnings. Can you fix those?

**Comment by Sommerbo** [comment]
It also have "var" datatype when wiki says not to use weak typing.

**Comment by sboigelot** [comment]
@Sommerbo Do we have a stylecop cnfig for that? It's hard to find as my vs auto replace them.

**Comment by sboigelot** [comment]
Ok so, the stylecop issues are fix, the merge conflict too, there is 5 stylecop issue left but I personally don't know how to fix them:

image

**Comment by Sommerbo** [comment]
@sboigelot Sorry no, since it isn't a style issue more of a coding convention.

**Comment by sboigelot** [comment]
@Sommerbo I replaced a lot of them by searching for var in the code, but there is still quite a bit in code not related to this PR

**Comment by Sommerbo** [comment]
Good start!

**Comment by vogonistic** [comment]
There are 5 StyleCop errors (SA1201) that I'm ignoring, just because this PR has been around for a while now and there has never been any big issues with it, just bad timing.

If you were to make a PR that fixes them, I'd take it in a heartbeat.

**Comment by sboigelot** [comment]
@vogonistic Great, thanks!

> If you were to make a PR that fixes them, I'd take it in a heartbeat.

I'll have a look during my break

**Comment by Dormanil** [comment]
@sboigelot I already fixed those in #1039

**Comment by sboigelot** [comment]
@Dormanil cool, thanks, I didn't realize it took into account the static properties! I learned!

**Pull request merged**